In [1]:
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning) # elimina los errres de que van a cambiar

# importamos las librerías que necesitamos
# Tratamiento de datos
# -----------------------------------------------------------------------
import pandas as pd
import numpy as np

# Imputación de nulos usando métodos avanzados estadísticos
# -----------------------------------------------------------------------
from sklearn.impute import SimpleImputer
from sklearn.experimental import enable_iterative_imputer
from sklearn.impute import IterativeImputer
from sklearn.impute import KNNImputer

# Librerías de visualización
# -----------------------------------------------------------------------
import seaborn as sns
import matplotlib.pyplot as plt
# Configuración
# -----------------------------------------------------------------------
pd.set_option('display.max_columns', None) # para poder visualizar todas las columnas de los DataFrames

from word2number import w2n

In [2]:
df = pd.read_csv('DisneyMoviesDataset.csv', index_col=0)

In [3]:
df.sample(10)

title  \
76            That Darn Cat!   
413                Frozen II   
113       Herbie Rides Again   
47                Jungle Cat   
128     Escape from the Dark   
398                     Coco   
33   Westward Ho the Wagons!   
317   Race to Witch Mountain   
264            Freaky Friday   
14       So Dear to My Heart   

                                    Production company  \
76                             Walt Disney Productions   
413  ['Walt Disney Pictures', 'Walt Disney Animatio...   
113                            Walt Disney Productions   
47                             Walt Disney Productions   
128                            Walt Disney Productions   
398  ['Walt Disney Pictures', 'Pixar Animation Stud...   
33                             Walt Disney Productions   
317             ['Walt Disney Pictures', 'Gunn Films']   
264             ['Walt Disney Pictures', 'Gunn Films']   
14                             Walt Disney Productions   

                                          Release date Running time  \
76                                ['December 2, 1965']  116 minutes   
413  ['November 7, 2019 ( Dolby Theatre )', 'Novemb...  103 minutes   
113                                   ['June 6, 1974']   88 minutes   
47                                 ['August 10, 1960']   69 minutes   
128  ['May 26, 1976', '(UK)', 'March 11, 1977', '(U...  104 minutes   
398  ['October 20, 2017 ( Morelia )', 'November 22,...  105 minutes   
33                               ['December 20, 1956']      90 min.   
317  ['March 11, 2009 (Egypt)', 'March 13, 2009 (Un...   98 minutes   
264  ['August 4, 2003 ( Los Angeles )', 'August 6, ...   97 minutes   
14   ['November 29, 1948 (Chicago, Illinois)', 'Jan...   82 minutes   

                                 Country Language  Running time (int)  \
76                         United States  English               116.0   
413                        United States  English               103.0   
113                        United States  English                88.0   
47                         United States  English                69.0   
128  ['United Kingdom', 'United States']  English               104.0   
398                        United States  English               105.0   
33                         United States  English                90.0   
317                        United States  English                98.0   
264                        United States  English                97.0   
14                         United States  English                82.0   

     Budget (float)  Box office (float) Release date (datetime)  imdb  \
76              NaN        2.806822e+07              1965-12-02   6.8   
413     150000000.0        1.450000e+09              2019-11-07   6.9   
113             NaN        3.822900e+07              1974-06-06   5.7   
47              NaN        2.300000e+06              1960-08-10   7.4   
128             NaN                 NaN              1976-05-26   NaN   
398     175000000.0        8.071000e+08              2017-10-20   8.4   
33              NaN        2.750000e+06              1956-12-20   5.8   
317      50000000.0        1.064000e+08              2009-03-11   5.7   
264      26000000.0        1.608000e+08              2003-08-04   6.2   
14        1500000.0        3.700000e+06              1948-11-29   6.8   

     metascore rotten_tomatoes                               Directed by  \
76        71.0             94%                          Robert Stevenson   
413       64.0             77%            ['Chris Buck', 'Jennifer Lee']   
113       49.0             80%                          Robert Stevenson   
47         NaN             NaN                               James Algar   
128        NaN             NaN                           Charles Jarrott   
398       81.0             97%                               Lee Unkrich   
33         NaN             NaN                          William Beaudine   
317       52.0             43%   

In [4]:
df.shape

(432, 31)

In [5]:
df.tail()

title  \
427                      Soul   
428  Raya and the Last Dragon   
429                   Cruella   
430             Jungle Cruise   
431     The Beatles: Get Back   

                                    Production company  \
427  ['Walt Disney Pictures', 'Pixar Animation Stud...   
428  ['Walt Disney Pictures', 'Walt Disney Animatio...   
429  ['Walt Disney Pictures', 'Gunn Films', 'Marc P...   
430  ['Walt Disney Pictures', 'Davis Entertainment'...   
431                                                NaN   

                                          Release date Running time  \
427  ['October 15, 2020 ( Rome Film Festival )', 'N...          NaN   
428                                 ['March 12, 2021']          NaN   
429                   ['May 28, 2021 (United States)']          NaN   
430                  ['July 30, 2021 (United States)']          NaN   
431                                 ['27 August 2021']          NaN   

                                               Country Language  \
427                                      United States  English   
428                                      United States  English   
429                                      United States  English   
430                                      United States  English   
431  ['United Kingdom', 'New Zealand', 'United Stat...  English   

     Running time (int)  Budget (float)  Box office (float)  \
427                 NaN     150000000.0                 NaN   
428                 NaN             NaN                 NaN   
429                 NaN             NaN                 NaN   
430                 NaN             NaN                 NaN   
431                 NaN             NaN                 NaN   

    Release date (datetime)  imdb  metascore rotten_tomatoes  \
427              2020-10-15   7.0       53.0             45%   
428              2021-03-12   NaN        NaN             NaN   
429              2021-05-28   NaN        NaN             NaN   
430              2021-07-30   NaN        NaN             NaN   
431              2021-08-27   8.0        NaN             NaN   

                              Directed by  \
427                           Pete Docter   
428  ['Don Hall', 'Carlos López Estrada']   
429                       Craig Gillespie   
430                    Jaume Collet-Serra   
431                         Peter Jackson   

                                           Produced by  \
427                                        Dana Murray   
428                ['Osnat Shurer', 'Peter Del Vecho']   
429      ['Kristin Burr', 'Andrew Gunn', 'Marc Platt']   
430  ['John Davis', 'John Fox', 'Beau Flynn', 'Dway...   
431  ['Peter Jackson', 'Clare Olssen', 'Jonathan Cl...   

                      Written by  \
427                          NaN   
428  ['Qui Nguyen', 'Adele Lim']   
429                          NaN   
430                          NaN   
431                          NaN   

                                              Based on  \
427                                                NaN   
428                                                NaN   
429  ["Disney's One Hundred and One Dalmatians by B...   
430             ['Jungle Cruise', 'by', 'Walt Disney']   
431                                                NaN   

                                              Starring  \
427  ['Jamie Foxx', 'Tina Fey', 'Questlove', 'Phyli...   
428                  ['Kelly Marie Tran', 'Awkwafina']   
429  ['Emma Stone', 'Emma Thompson', 'Paul Walter H...   
430  ['Dwayne Johnson', 'Emily Blunt', 'Édgar Ramír...   
431                                        The Beatles   

                                 Music by  \
427      ['Trent Reznor', 'Atticus Ross']   
428                                   NaN   
429                                   NaN   
430  ['James Newton Howard', 'Metallica']   
431                           The Beatles   

                                 Distributed by         Budget Box office  \

In [6]:
#Creamos una función para entender el data frame


def entender_datos(df):
    
    # Encabezado
    print("INFO SOBRE NUESTROS DATOS")
    print("============================================================================")
    
    # Información general
    print(f"El número de filas es: {df.shape[0]}")
    print(f"El número de columnas es: {df.shape[1]}")
    print("============================================================================")
    
    # Descripción del DataFrame
    print("DESCRIPCIÓN ESTADÍSTICA")
    print(df.describe().T)
    print("============================================================================")
    
    # Columnas del DataFrame
    print("COLUMNAS")
    print(df.columns)
    print("============================================================================")
    
    # Filas duplicadas
    print(f"Número de filas duplicadas: {df.duplicated().sum()}")
    print("============================================================================")
    
    # Valores nulos totales
    print(f"Total de valores nulos: {df.isna().sum().sum()}")
    print("============================================================================")
    
    # Valores nulos por columna
    print("VALORES NULOS POR COLUMNA")
    print(df.isna().sum())
    print("============================================================================")

    #Porcentaje de valores nulos por columna
    print("PORCENTAJE NULOS POR COLUMNA")
    print(round(df.isna().sum() / df.shape[0] * 100,2))
    print("============================================================================")

    # Tipos de datos
    print("TIPOS DE DATOS")
    print(df.dtypes.value_counts())
    print("============================================================================")
    
    # Información del DataFrame
    print("INFORMACIÓN DETALLADA")
    df.info()
    print("============================================================================")



In [7]:
df['Country'].unique()

array(['United States', "['United States', 'United Kingdom']",
       "['United Kingdom', 'United States']", 'United States, Mexico',
       "['United States', 'Canada']", nan,
       "['United States', 'Australia']", "['United States', 'Norway']",
       'Australia', "['Norway', 'Sweden', 'United States']",
       "['United States', 'Austria']", "['United States', 'France']",
       "['United States', 'United Kingdom', 'Germany']",
       "['United States', 'United Kingdom', 'France']",
       "['Canada', 'Malaysia', 'United States']",
       "['France', 'United Kingdom', 'Germany', 'United States']",
       "['Germany', 'Austria', 'Italy', 'Spain', 'United Kingdom']",
       'Russia', 'France', 'India', 'Germany',
       "['United States', 'France', 'United Kingdom']",
       "['France', 'United States']",
       "['Australia', 'United Kingdom', 'United States']",
       "['Spain', 'Italy']", "['United States']",
       "['United States', 'China', 'France']",
       "['United Kingdom

In [8]:
entender_datos(df)

INFO SOBRE NUESTROS DATOS
El número de filas es: 432
El número de columnas es: 31
DESCRIPCIÓN ESTADÍSTICA
                    count          mean           std    min         25%  \
Running time (int)  422.0  9.730569e+01  1.895949e+01   40.0        86.0   
Budget (float)      273.0  6.358861e+07  7.163732e+07  150.0  10000000.0   
Box office (float)  355.0  1.674014e+08  2.749517e+08    7.7   9850000.0   
imdb                416.0  6.545433e+00  9.609374e-01    1.5         6.0   
metascore           283.0  6.148763e+01  1.656266e+01   18.0        49.0   

                           50%           75%           max  
Running time (int)        96.0  1.067500e+02  1.680000e+02  
Budget (float)      30000000.0  1.000000e+08  4.106000e+08  
Box office (float)  42900000.0  1.865500e+08  1.657000e+09  
imdb                       6.6  7.200000e+00  8.700000e+00  
metascore                 61.0  7.300000e+01  9.900000e+01  
COLUMNAS
Index(['title', 'Production company', 'Release date', 'Running

 0   title                    432 non-null    object  OK  ***SI***

 1   Production company       392 non-null    object  Lista (ver como hacer) (ver como hacer) Hacer explode. USA --- Nulos ***SI***

 2   Release date             431 non-null    object  Tenemos lista hay otra columna con este dato Podemos hacer una nueva columna con el lugar de estreno dividido entre parentesis ***SI*** LUGAR DE ESTENO ()

 3   Running time             422 non-null    object  ***ELIMINAR COLUMNA***

 4   Country                  428 non-null    object  Lista (ver como hacer) Hacer explode ***SI***

 5   Language                 430 non-null    object  Lista (ver como hacer) Hacer explode ***SI***

 6   Running time (int)       422 non-null    float64 OK coger media para nulos ***SI***

 7   Budget (float)           273 non-null    float64 Ok gestionar nulos ***SI***

 8   Box office (float)       355 non-null    float64 ok gestionar nulos ***SI*** (GANADO EN TAQUILLA)

 9   Release date (datetime)  428 non-null    object  ok los nulos como una categoria nueva ***SI*** 

 10  imdb                     416 non-null    float64 ok  media mediana para nulos ***SI*** 

 11  metascore                283 non-null    float64  ***ELIMINAR COLUMNA***

 12  rotten_tomatoes          361 non-null    object  ***ELIMINAR COLUMNA***

 13  Directed by              431 non-null    object  tiene listas, quededarnos con el primero pero quitar las categorias que no vamos a usar ***SI*** 

 14  Produced by              422 non-null    object nos quedamos con el primero ***SI*** 

 15  Written by               203 non-null    object idem directed by ***ELIMINAR COLUMNA***

 16  Based on                 211 non-null    object lista convertir string, limpieza de caracteres ***SI*** 

 17  Starring                 398 non-null    object explode es una lista, nos quedamos con todo ***ELIMINAR COLUMNA***

 18  Music by                 422 non-null    object es una lista, nos quedamos con el primero ***ELIMINAR COLUMNA***

 19  Distributed by           429 non-null    object es una lista, nos quedamos con el primero ***SI***

 20  Budget                   276 non-null    object ***ELIMINAR COLUMNA***

 21  Box office               359 non-null    object ***ELIMINAR COLUMNA***

 22  Story by                 139 non-null    object es una lista, nos quedamos con el primero ***ELIMINAR COLUMNA***

 23  Narrated by              55 non-null     object ***ELIMINAR COLUMNA***

 24  Cinematography           342 non-null    object es una lista, nos quedamos con el primero, y nombres ***ELIMINAR COLUMNA***

 25  Edited by                395 non-null    object es una lista, nos quedamos con el primero, y nombres ***ELIMINAR COLUMNA***

 26  Screenplay by            191 non-null    object tiene listas, quededarnos con el primero pero quitar las categorias que no vamos a usar ***ELIMINAR COLUMNA***

 27  Production companies     39 non-null     object ***ELIMINAR COLUMNA***

 28  Adaptation by            1 non-null      object ***ELIMINAR COLUMNA***

 29  Traditional              1 non-null      object ***ELIMINAR COLUMNA***

 30  Simplified               1 non-null      object ***ELIMINAR COLUMNA***

In [9]:
df['Distributed by'].unique()

array([nan, 'RKO Radio Pictures',
       "['Walt Disney Productions', 'RKO Radio Pictures']",
       'United Artists', 'RKO Radio Pictures, Inc.',
       'Buena Vista Distribution',
       'Buena Vista Film Distribution Co., Inc.',
       'Buena Vista Film Distribution',
       'Buena Vista Pictures Distribution',
       'Buena Vista Distribution Co., Inc.',
       "['Buena Vista Distribution', '(US)', 'British Empire Films', '(Australia)']",
       "['Paramount Pictures (North America)', '(North America)', 'Buena Vista International Distribution (International)', '(International)']",
       "['Paramount Pictures', 'Buena Vista International Distribution']",
       "['PSO Productions (outside North America)', 'Buena Vista Distribution']",
       'Buena Vista Pictures',
       "['Buena Vista Pictures', '(North America and other select territories)', 'MDP Worldwide', '(all other territories)']",
       "['Buena Vista Pictures', '(United States)', 'Trimark Pictures (International)']",
   

In [10]:
df.head()


title       Production company  \
0          Academy Award Review of  Walt Disney Productions   
1  Snow White and the Seven Dwarfs  Walt Disney Productions   
2                        Pinocchio  Walt Disney Productions   
3                         Fantasia  Walt Disney Productions   
4             The Reluctant Dragon  Walt Disney Productions   

                                        Release date  \
0                                   ['May 19, 1937']   
1  ['December 21, 1937 ( Carthay Circle Theatre ,...   
2  ['February 7, 1940 ( Center Theatre )', 'Febru...   
3                              ['November 13, 1940']   
4                                  ['June 20, 1941']   

                           Running time        Country Language  \
0  41 minutes (74 minutes 1966 release)  United States  English   
1                            83 minutes  United States  English   
2                            88 minutes  United States  English   
3                           126 minutes  United States  English   
4                            74 minutes  United States  English   

   Running time (int)  Budget (float)  Box office (float)  \
0                41.0             NaN                 NaN   
1                83.0       1490000.0         418000000.0   
2                88.0       2600000.0         164000000.0   
3               126.0       2280000.0          83300000.0   
4                74.0        600000.0            960000.0   

  Release date (datetime)  imdb  metascore rotten_tomatoes  \
0              1937-05-19   7.2        NaN             NaN   
1              1937-12-21   7.6       95.0             NaN   
2              1940-02-07   7.4       99.0            100%   
3              1940-11-13   7.8       96.0             95%   
4              1941-06-20   6.9        NaN             67%   

                                         Directed by  \
0                                                NaN   
1  ['David Hand (supervising)', 'William Cottrell...   
2  ['Ben Sharpsteen', 'Hamilton Luske', 'Bill Rob...   
3  ['Samuel Armstrong', 'James Algar', 'Bill Robe...   
4  ['Alfred Werker', '(live action)', 'Hamilton L...   

                         Produced by  \
0                                NaN   
1                        Walt Disney   
2                        Walt Disney   
3  ['Walt Disney', 'Ben Sharpsteen']   
4                        Walt Disney   

                                          Written by  \
0                                                NaN   
1  ['Ted Sears', 'Richard Creedon', 'Otto England...   
2                                                NaN   
3                                                NaN   
4  ['Live-action:', 'Ted Sears', 'Al Perkins', 'L...   

                                            Based on  \
0                                                NaN   
1         ['Snow White', 'by The', 'Brothers Grimm']   
2  ['The Adventures of Pinocchio', 'by', 'Carlo C...   
3                                                NaN   
4                                                NaN   

                                            Starring  \
0                                                NaN   
1  ['Adriana Caselotti', 'Lucille La Verne', 'Har...   
2  ['Cliff Edwards', 'Dickie Jones', 'Christian R...   
3              ['Leopold Stokowski', 'Deems Taylor']   
4  ['Robert Benchley', 'Frances Gifford', 'Buddy ...   

                                            Music by  \
0                                                NaN   
1  ['Frank Churchill', 'Paul Smith', 'Leigh Harli...   
2                 ['Leigh Harline', 'Paul J. Smith']   
3                                        See program   
4                 ['Frank Churchill', 'Larry Morey']   

                                      Distributed by         Budget  \
0                                                NaN            NaN   
1                                 RKO Radio Pictures  $1.49 million   
2                              

In [11]:
df2_cat = df.select_dtypes(include = "object")


# creamos una variable con los nombres de las columnas del DataFrame de las variables categóricas utilizando el método '.columns'
columnas_cat = df2_cat.columns
print(f"Las columnas del DataFrame de variables categóricas son {columnas_cat}")

# empezamos a iterar por cada una de las columnas para sacar sus valores únicos y sus frecuencias
for columna in columnas_cat:
    print(f" \n----------- ESTAMOS ANALIZANDO LA COLUMNA: '{columna.upper()}' -----------\n")
    print(f"Sus valores únicos son: {df2_cat[columna].unique()}\n")
    print(f"Las frecuencias de los valores únicos de las categorías son: {df2_cat[columna].value_counts()} ")

Las columnas del DataFrame de variables categóricas son Index(['title', 'Production company', 'Release date', 'Running time',
       'Country', 'Language', 'Release date (datetime)', 'rotten_tomatoes',
       'Directed by', 'Produced by', 'Written by', 'Based on', 'Starring',
       'Music by', 'Distributed by', 'Budget', 'Box office', 'Story by',
       'Narrated by', 'Cinematography', 'Edited by', 'Screenplay by',
       'Production companies', 'Adaptation by', 'Traditional', 'Simplified'],
      dtype='object')
 
----------- ESTAMOS ANALIZANDO LA COLUMNA: 'TITLE' -----------

Sus valores únicos son: ['Academy Award Review of' 'Snow White and the Seven Dwarfs' 'Pinocchio'
 'Fantasia' 'The Reluctant Dragon' 'Dumbo' 'Bambi' 'Saludos Amigos'
 'Victory Through Air Power' 'The Three Caballeros' 'Make Mine Music'
 'Song of the South' 'Fun and Fancy Free' 'Melody Time'
 'So Dear to My Heart' 'The Adventures of Ichabod and Mr. Toad'
 'Cinderella' 'Treasure Island' 'Alice in Wonderland'
 'The

Sus valores únicos son: ["['May 19, 1937']"
 "['December 21, 1937 ( Carthay Circle Theatre , Los Angeles , CA )', 'February 4, 1938 (United States)']"
 "['February 7, 1940 ( Center Theatre )', 'February 23, 1940 (United States)']"
 "['November 13, 1940']" "['June 20, 1941']"
 "['October 23, 1941 (New York City)', 'October 31, 1941 (U.S.)']"
 "['August 9, 1942 (World Premiere-London)', 'August 13, 1942 (Premiere-New York City)', 'August 21, 1942 (U.S.)']"
 "['August 24, 1942 (World Premiere-Rio de Janeiro)', 'February 6, 1943 (U.S. Premiere-Boston)', 'February 19, 1943 (U.S.)']"
 "['July 17, 1943']"
 "['December 21, 1944 (Mexico City)', 'February 3, 1945 (US)']"
 "['April 20, 1946 (Premiere-New York City)', 'August 15, 1946 (U.S.)']"
 "['November 12, 1946 (Premiere: Atlanta, Georgia)', 'November 20, 1946']"
 "['September 27, 1947']" 'May 27, 1948'
 "['November 29, 1948 (Chicago, Illinois)', 'January 19, 1949 (Indianapolis, Indiana)']"
 "['October 5, 1949']"
 "['February 15, 1950 (Boston

### LIMPIEZA

In [12]:
#Borramos columnas innecesarias para el análsis
columnas_a_eliminar = ['Release date','Running time', 'metascore', 'rotten_tomatoes', 'Written by', 'Starring', 'Music by', 'Budget', 'Box office', 'Story by', 'Narrated by', 'Cinematography', 'Edited by', 'Screenplay by', 'Production companies', 'Adaptation by', 'Traditional', 'Simplified']
def eliminar_columnas(df, lista):
    df.drop( columns = lista, inplace= True)
eliminar_columnas(df,columnas_a_eliminar) 



In [13]:
df.shape

(432, 13)

In [14]:
df.columns


Index(['title', 'Production company', 'Country', 'Language',
       'Running time (int)', 'Budget (float)', 'Box office (float)',
       'Release date (datetime)', 'imdb', 'Directed by', 'Produced by',
       'Based on', 'Distributed by'],
      dtype='object')

In [15]:
#hacemos una variable de nuevas columnas que hará que las columnas aparezcan en minuscula y que remplace el espacio por comas
new_columns = {column : column.lower().replace(' ', '_') for column in df}

#renombramos las columnas antiguas por las nuevas
df.rename(columns = new_columns, inplace= True)

# comprobamos que se ha realizado
df.sample(10)

title  \
96                         The Boatniks   
264                       Freaky Friday   
41                       The Shaggy Dog   
49                Swiss Family Robinson   
76                       That Darn Cat!   
85         Charlie, the Lonesome Cougar   
142                The Omega Connection   
427                                Soul   
116  The Island at the Top of the World   
131                        The Rescuers   

                                    production_company        country  \
96                             Walt Disney Productions  United States   
264             ['Walt Disney Pictures', 'Gunn Films']  United States   
41                             Walt Disney Productions  United States   
49                             Walt Disney Productions  United States   
76                             Walt Disney Productions  United States   
85                             Walt Disney Productions  United States   
142                            Walt Disney Productions  United States   
427  ['Walt Disney Pictures', 'Pixar Animation Stud...  United States   
116                            Walt Disney Productions  United States   
131                            Walt Disney Productions  United States   

                                              language  running_time_(int)  \
96                                             English               100.0   
264                                            English                97.0   
41                                             English               104.0   
49                                             English               126.0   
76                                             English               116.0   
85                                             English                75.0   
142                                            English               100.0   
427                                            English                 NaN   
116  ['English', 'Swedish', 'Norwegian', 'Danish', ...                93.0   
131                                            English                77.0   

     budget_(float)  box_office_(float) release_date_(datetime)  imdb  \
96              NaN          18607492.0              1970-07-01   5.5   
264      26000000.0         160800000.0              2003-08-04   6.2   
41        1000000.0          12300000.0              1959-03-19   4.4   
49        4000000.0          40000000.0              1960-12-21   7.2   
76              NaN          28068222.0              1965-12-02   6.8   
85              NaN                 NaN              1967-10-18   6.9   
142             NaN                 NaN              1979-12-21   NaN   
427     150000000.0                 NaN              2020-10-15   7.0   
116       8000000.0          10000000.0              1974-12-20   6.3   
131       7500000.0         169000000.0              1977-06-22   6.9   

                                           directed_by  \
96                                        Norman Tokar   
264                                        Mark Waters   
41                                      Charles Barton   
49                    ['John McKimson', 'Ken Annakin']   
76                                    Robert Stevenson   
85                     ['Rex Allen', 'Winston Hibler']   
142                                      Robert Clouse   
427                                        Pete Docter   
116                                   Robert Stevenson   
131  ['Wolfgang Reitherman', 'John Lounsbery', 'Art...   

                                           produced_by  \
96                                          Ron Miller   
264                                        Andrew Gunn   
41                       ['Walt Disney', 'Bill Walsh']   
49      ['Bill Anderson', 'Basil Keys', 'Walt Disney']   
76                       ['Walt Disney', 'Bill Walsh']   
85   ['Walt Disney', 'Winston Hibler', 'Charles L. ...   
142                                       Jan Wi

In [16]:
# quitamos int, float y datatime del titulo
df.columns = [col.replace('_(int)', '') for col in df.columns]
df.columns = [col.replace('_(float)', '') for col in df.columns]
df.columns = [col.replace('_(datetime)', '') for col in df.columns]


In [17]:
df.head(1)

title       production_company        country language  \
0  Academy Award Review of  Walt Disney Productions  United States  English   

   running_time  budget  box_office release_date  imdb directed_by  \
0          41.0     NaN         NaN   1937-05-19   7.2         NaN   

  produced_by based_on distributed_by  
0         NaN      NaN            NaN

In [18]:
df.columns

Index(['title', 'production_company', 'country', 'language', 'running_time',
       'budget', 'box_office', 'release_date', 'imdb', 'directed_by',
       'produced_by', 'based_on', 'distributed_by'],
      dtype='object')

 0   title                    432 non-null    object  OK  ***SI***

 1   Production company       392 non-null    object  Lista (ver como hacer) (ver como hacer) Hacer explode. USA --- Nulos ***SI***

 2   Release date             431 non-null    object  Tenemos lista hay otra columna con este dato Podemos hacer una nueva columna con el lugar de estreno dividido entre parentesis ***SI*** LUGAR DE ESTENO ()
 
 4   Country                  428 non-null    object  Lista (ver como hacer) Hacer explode ***SI***

 5   Language                 430 non-null    object  Lista (ver como hacer) Hacer explode ***SI***

 6   Running time (int)       422 non-null    float64 OK coger media para nulos ***SI***

 7   Budget (float)           273 non-null    float64 Ok gestionar nulos ***SI***

 8   Box office (float)       355 non-null    float64 ok gestionar nulos ***SI*** (GANADO EN TAQUILLA)

 9   Release date (datetime)  428 non-null    object  ok los nulos como una categoria nueva ***SI*** 

 10  imdb                     416 non-null    float64 ok  media mediana para nulos ***SI*** 

13  Directed by              431 non-null    object  tiene listas, quededarnos con el primero pero quitar las categorias que no vamos a usar ***SI*** 

14  Produced by              422 non-null    object nos quedamos con el primero ***SI*** 
  
16  Based on                 211 non-null    object lista convertir string, limpieza de caracteres ***SI*** 

19  Distributed by           429 non-null    object es una lista, nos quedamos con el primero ***SI***


 0   title                    432 non-null    object  OK  ***SI***

 1   Production company       392 non-null    object  Lista (ver como hacer) (ver como hacer) Hacer explode. USA --- Nulos ***SI***

 2   Release date             431 non-null    object  Tenemos lista hay otra columna con este dato Podemos hacer una nueva columna con el lugar de estreno dividido entre parentesis ***SI*** LUGAR DE ESTENO ()
 
 4   Country                  428 non-null    object  Lista (ver como hacer) Hacer explode ***SI***

 5   Language                 430 non-null    object  Lista (ver como hacer) Hacer explode ***SI***

 6   Running time (int)       422 non-null    float64 OK coger media para nulos ***SI***

 7   Budget (float)           273 non-null    float64 Ok gestionar nulos ***SI***

 8   Box office (float)       355 non-null    float64 ok gestionar nulos ***SI*** (GANADO EN TAQUILLA)

 9   Release date (datetime)  428 non-null    object  ok los nulos como una categoria nueva ***SI*** 

 10  imdb                     416 non-null    float64 ok  media mediana para nulos ***SI*** 

13  Directed by              431 non-null    object  tiene listas, quededarnos con el primero pero quitar las categorias que no vamos a usar ***SI*** 

14  Produced by              422 non-null    object nos quedamos con el primero ***SI*** 
  
16  Based on                 211 non-null    object lista convertir string, limpieza de caracteres ***SI*** 

19  Distributed by           429 non-null    object es una lista, nos quedamos con el primero ***SI***


Index(['title', 'production_company', 'country', 'language', 'running_time',
       'budget', 'box_office', 'release_date', 'imdb', 'directed_by',
       'produced_by', 'based_on', 'distributed_by'],
      dtype='object')

In [19]:
df.columns

Index(['title', 'production_company', 'country', 'language', 'running_time',
       'budget', 'box_office', 'release_date', 'imdb', 'directed_by',
       'produced_by', 'based_on', 'distributed_by'],
      dtype='object')

In [20]:
# Lista de columnas a transformar con explode
columns_to_explode = ['production_company', 'country', 'language']

# Iterar sobre cada columna y aplicar las transformaciones
for column in columns_to_explode:
    if column in df.columns:
        # Convertir todos los valores a cadenas para evitar errores
        df[column] = df[column].astype(str)
        # Quitar corchetes, eliminar comillas y dividir por comas
        df[column] = df[column].str.strip("[]").str.replace("'", "").str.split(", ")
        # Aplicar explode
        df = df.explode(column)
        # Limpiar espacios en blanco alrededor de los elementos
        df[column] = df[column].str.strip()

# Revisar la forma del DataFrame final
print(f"Forma después de aplicar explode a todas las columnas: {df.shape}")

# Mostrar las primeras filas del DataFrame final usando display
display(df.head())


Forma después de aplicar explode a todas las columnas: (974, 13)


title       production_company        country  \
0          Academy Award Review of  Walt Disney Productions  United States   
1  Snow White and the Seven Dwarfs  Walt Disney Productions  United States   
2                        Pinocchio  Walt Disney Productions  United States   
3                         Fantasia  Walt Disney Productions  United States   
4             The Reluctant Dragon  Walt Disney Productions  United States   

  language  running_time     budget   box_office release_date  imdb  \
0  English          41.0        NaN          NaN   1937-05-19   7.2   
1  English          83.0  1490000.0  418000000.0   1937-12-21   7.6   
2  English          88.0  2600000.0  164000000.0   1940-02-07   7.4   
3  English         126.0  2280000.0   83300000.0   1940-11-13   7.8   
4  English          74.0   600000.0     960000.0   1941-06-20   6.9   

                                         directed_by  \
0                                                NaN   
1  ['David Hand (supervising)', 'William Cottrell...   
2  ['Ben Sharpsteen', 'Hamilton Luske', 'Bill Rob...   
3  ['Samuel Armstrong', 'James Algar', 'Bill Robe...   
4  ['Alfred Werker', '(live action)', 'Hamilton L...   

                         produced_by  \
0                                NaN   
1                        Walt Disney   
2                        Walt Disney   
3  ['Walt Disney', 'Ben Sharpsteen']   
4                        Walt Disney   

                                            based_on  \
0                                                NaN   
1         ['Snow White', 'by The', 'Brothers Grimm']   
2  ['The Adventures of Pinocchio', 'by', 'Carlo C...   
3                                                NaN   
4                                                NaN   

                                      distributed_by  
0                                                NaN  
1                                 RKO Radio Pictures  
2                                 RKO Radio Pictures  
3  ['Walt Disney Productions', 'RKO Radio Pictures']  
4                                 RKO Radio Pictures

### NULOS

In [21]:
#Exploracion columnas 

def eda_columnas(df):
    
    # Separar columnas categóricas y numéricas
    columnas_categoricas = df.select_dtypes(include=['object', 'category']).columns
    columnas_numericas = df.select_dtypes(include=['number']).columns
    
    display("Columnas categóricas:", columnas_categoricas)
    display("Columnas numéricas:", columnas_numericas)
    
    # Exploración de columnas categóricas
    for columna in columnas_categoricas:
        display(f"Exploración de la columna categórica: {columna}")
        datos_columna = df[columna]  # Extrae los datos de la columna específica

        # Descripción estadística
        display("Descripción estadística:")
        display(datos_columna.describe(include='all'))
        
        # Valores nulos
        display("Conteo de valores nulos:")
        display(datos_columna.isnull().sum())
        
        # Valores duplicados
        display("Número de filas duplicadas:")
        display(df.duplicated(subset=[columna]).sum())
       
        # Valores únicos
        display("Conteo de valores únicos:")
        display(datos_columna.nunique())
        
        # Conteo de cada valor
        display("Conteo de valores:")
        display(datos_columna.value_counts())
        
        # Valor más frecuente (moda)
        if not datos_columna.mode().empty:
            display("Valor más frecuente (moda):")
            display(datos_columna.mode().iloc[0])
        
        display("-" * 30)

    # Exploración de columnas numéricas
    for columna in columnas_numericas:
        display(f"Exploración de la columna numérica: {columna}")
        datos_columna = df[columna]  # Extrae los datos de la columna específica

        # Descripción estadística
        display("Descripción estadística:")
        display(datos_columna.describe())
        
        # Valores nulos
        display("Conteo de valores nulos:")
        display(datos_columna.isnull().sum())
        
        # Valores duplicados
        display("Número de filas duplicadas:")
        display(df.duplicated(subset=[columna]).sum())
       
        # Valores únicos
        display("Conteo de valores únicos:")
        display(datos_columna.nunique())
        
        display("-" * 30)

# Llamada a la función con tu DataFrame modificado llamado 'df'
eda_columnas(df)


'Columnas categóricas:'

Index(['title', 'production_company', 'country', 'language', 'release_date',
       'directed_by', 'produced_by', 'based_on', 'distributed_by'],
      dtype='object')

'Columnas numéricas:'

Index(['running_time', 'budget', 'box_office', 'imdb'], dtype='object')

'Exploración de la columna categórica: title'

'Descripción estadística:'

count               974
unique              419
top       Born in China
freq                 36
Name: title, dtype: object

'Conteo de valores nulos:'

np.int64(0)

'Número de filas duplicadas:'

np.int64(555)

'Conteo de valores únicos:'

419

'Conteo de valores:'

title
Born in China                  36
African Cats                   18
Earth                          16
Around the World in 80 Days    16
Saving Mr. Banks               15
                               ..
Clouds                          1
Elephant                        1
Dolphin Reef                    1
The Reluctant Dragon            1
Fantasia                        1
Name: count, Length: 419, dtype: int64

'Valor más frecuente (moda):'

'Born in China'

'------------------------------'

'Exploración de la columna categórica: production_company'

'Descripción estadística:'

count                      974
unique                     214
top       Walt Disney Pictures
freq                       236
Name: production_company, dtype: object

'Conteo de valores nulos:'

np.int64(0)

'Número de filas duplicadas:'

np.int64(760)

'Conteo de valores únicos:'

214

'Conteo de valores:'

production_company
Walt Disney Pictures         236
Walt Disney Productions      186
nan                           48
Disneynature                  33
Pixar Animation Studios       19
                            ... 
Jason T. Reed Productions      1
Good Fear Productions          1
Davis Entertainment            1
Seven Bucks Productions        1
Flynn Picture Company          1
Name: count, Length: 214, dtype: int64

'Valor más frecuente (moda):'

'Walt Disney Pictures'

'------------------------------'

'Exploración de la columna categórica: country'

'Descripción estadística:'

count               974
unique               18
top       United States
freq                760
Name: country, dtype: object

'Conteo de valores nulos:'

np.int64(0)

'Número de filas duplicadas:'

np.int64(956)

'Conteo de valores únicos:'

18

'Conteo de valores:'

country
United States     760
United Kingdom     61
France             50
India              21
Canada             15
Australia          12
China              12
nan                11
Germany             7
Norway              6
Sweden              4
Austria             4
Russia              3
Mexico              2
Italy               2
Spain               2
Malaysia            1
New Zealand         1
Name: count, dtype: int64

'Valor más frecuente (moda):'

'United States'

'------------------------------'

'Exploración de la columna categórica: language'

'Descripción estadística:'

count         974
unique         22
top       English
freq          840
Name: language, dtype: object

'Conteo de valores nulos:'

np.int64(0)

'Número de filas duplicadas:'

np.int64(952)

'Conteo de valores únicos:'

22

'Conteo de valores:'

language
English             840
French               50
Hindi                21
Mandarin             12
Norwegian             7
Spanish               6
German                6
Mandarin Chinese      4
Telugu                4
nan                   4
Russian               3
Vietnamese            3
Gaelic                2
Portuguese            2
Inuktitut             2
Italian               2
Scots                 1
Irish                 1
Danish                1
Swedish               1
Icelandic             1
Luganda               1
Name: count, dtype: int64

'Valor más frecuente (moda):'

'English'

'------------------------------'

'Exploración de la columna categórica: release_date'

'Descripción estadística:'

count            950
unique           421
top       2016-08-12
freq              36
Name: release_date, dtype: object

'Conteo de valores nulos:'

np.int64(24)

'Número de filas duplicadas:'

np.int64(552)

'Conteo de valores únicos:'

421

'Conteo de valores:'

release_date
2016-08-12    36
2011-04-22    20
2007-10-10    16
2004-06-13    16
2013-10-20    15
              ..
2018-06-05     1
2018-11-05     1
2019-06-11     1
1937-12-21     1
1937-05-19     1
Name: count, Length: 421, dtype: int64

'Valor más frecuente (moda):'

'2016-08-12'

'------------------------------'

'Exploración de la columna categórica: directed_by'

'Descripción estadística:'

count          973
unique         282
top       Lu Chuan
freq            36
Name: directed_by, dtype: object

'Conteo de valores nulos:'

np.int64(1)

'Número de filas duplicadas:'

np.int64(691)

'Conteo de valores únicos:'

282

'Conteo de valores:'

directed_by
Lu Chuan                                                                                                                                                                                                     36
['Alastair Fothergill', 'Mark Linfield']                                                                                                                                                                     28
Robert Stevenson                                                                                                                                                                                             26
['Alastair Fothergill', 'Keith Scholey']                                                                                                                                                                     21
Frank Coraci                                                                                                                                                

'Valor más frecuente (moda):'

'Lu Chuan'

'------------------------------'

'Exploración de la columna categórica: produced_by'

'Descripción estadística:'

count             953
unique            286
top       Walt Disney
freq               40
Name: produced_by, dtype: object

'Conteo de valores nulos:'

np.int64(21)

'Número de filas duplicadas:'

np.int64(687)

'Conteo de valores únicos:'

286

'Conteo de valores:'

produced_by
Walt Disney                                     40
['Roy Conli', 'Brian Leith', 'Phil Chapman']    36
Ron Miller                                      19
['Alix Tidmarsh', 'Keith Scholey']              18
['Bill Badalato', 'Hal Lieberman']              16
                                                ..
James Whitaker                                   1
Osnat Shurer                                     1
['James C. Pratt', 'John Lasseter']              1
['Lawrence Edward Watkin', 'Walt Disney']        1
['Walt Disney', 'Ben Sharpsteen']                1
Name: count, Length: 286, dtype: int64

'Valor más frecuente (moda):'

'Walt Disney'

'------------------------------'

'Exploración de la columna categórica: based_on'

'Descripción estadística:'

count                                                   414
unique                                                  197
top       ['Around the World in Eighty Days', 'by', 'Jul...
freq                                                     16
Name: based_on, dtype: object

'Conteo de valores nulos:'

np.int64(560)

'Número de filas duplicadas:'

np.int64(776)

'Conteo de valores únicos:'

197

'Conteo de valores:'

based_on
['Around the World in Eighty Days', 'by', 'Jules Verne']                                           16
["Disney's Cinderella", 'Cendrillon by Charles Perrault']                                           8
['by']                                                                                              6
['James and the Giant Peach', 'by', 'Roald Dahl']                                                   6
['Treasure Island', 'by', 'Robert Louis Stevenson']                                                 6
                                                                                                   ..
['Oz', 'by', 'L. Frank Baum']                                                                       1
['A Wrinkle in Time', 'by', "Madeleine L'Engle"]                                                    1
["Pete's Dragon", 'by', 'Malcolm Marmorstein']                                                      1
['The BFG', 'by', 'Roald Dahl']                                          

'Valor más frecuente (moda):'

"['Around the World in Eighty Days', 'by', 'Jules Verne']"

'------------------------------'

'Exploración de la columna categórica: distributed_by'

'Descripción estadística:'

count                      965
unique                      33
top       Buena Vista Pictures
freq                       274
Name: distributed_by, dtype: object

'Conteo de valores nulos:'

np.int64(9)

'Número de filas duplicadas:'

np.int64(940)

'Conteo de valores únicos:'

33

'Conteo de valores:'

distributed_by
Buena Vista Pictures                                                                                                                         274
Buena Vista Distribution                                                                                                                     173
['Walt Disney Studios', 'Motion Pictures']                                                                                                   160
Walt Disney Studios Motion Pictures                                                                                                          115
Buena Vista Pictures Distribution                                                                                                             40
['Walt Disney Studios Motion Pictures', 'Shanghai Media Group', '(China)']                                                                    36
RKO Radio Pictures                                                                                                 

'Valor más frecuente (moda):'

'Buena Vista Pictures'

'------------------------------'

'Exploración de la columna numérica: running_time'

'Descripción estadística:'

count    947.000000
mean      97.139388
std       19.295291
min       40.000000
25%       83.000000
50%       96.000000
75%      107.000000
max      168.000000
Name: running_time, dtype: float64

'Conteo de valores nulos:'

np.int64(27)

'Número de filas duplicadas:'

np.int64(885)

'Conteo de valores únicos:'

88

'------------------------------'

'Exploración de la columna numérica: budget'

'Descripción estadística:'

count    7.220000e+02
mean     5.808164e+07
std      6.421383e+07
min      1.500000e+02
25%      1.000000e+07
50%      3.000000e+07
75%      9.000000e+07
max      4.106000e+08
Name: budget, dtype: float64

'Conteo de valores nulos:'

np.int64(252)

'Número de filas duplicadas:'

np.int64(869)

'Conteo de valores únicos:'

104

'------------------------------'

'Exploración de la columna numérica: box_office'

'Descripción estadística:'

count    8.340000e+02
mean     1.517666e+08
std      2.505806e+08
min      7.700000e+00
25%      1.643034e+07
50%      5.060000e+07
75%      1.632000e+08
max      1.657000e+09
Name: box_office, dtype: float64

'Conteo de valores nulos:'

np.int64(140)

'Número de filas duplicadas:'

np.int64(647)

'Conteo de valores únicos:'

326

'------------------------------'

'Exploración de la columna numérica: imdb'

'Descripción estadística:'

count    935.000000
mean       6.599893
std        0.997271
min        1.500000
25%        6.000000
50%        6.700000
75%        7.300000
max        8.700000
Name: imdb, dtype: float64

'Conteo de valores nulos:'

np.int64(39)

'Número de filas duplicadas:'

np.int64(924)

'Conteo de valores únicos:'

49

'------------------------------'

In [22]:
df.isnull().sum()

title                   0
production_company      0
country                 0
language                0
running_time           27
budget                252
box_office            140
release_date           24
imdb                   39
directed_by             1
produced_by            21
based_on              560
distributed_by          9
dtype: int64

In [23]:
# Porcentaje de valores nulos
print("Porcentaje de valores nulos por columna:")
display((df.isnull().sum() / len(df) * 100).round(2))
print("----"*10)

Porcentaje de valores nulos por columna:


title                  0.00
production_company     0.00
country                0.00
language               0.00
running_time           2.77
budget                25.87
box_office            14.37
release_date           2.46
imdb                   4.00
directed_by            0.10
produced_by            2.16
based_on              57.49
distributed_by         0.92
dtype: float64

----------------------------------------


In [24]:
df.head(1)

title       production_company        country language  \
0  Academy Award Review of  Walt Disney Productions  United States  English   

   running_time  budget  box_office release_date  imdb directed_by  \
0          41.0     NaN         NaN   1937-05-19   7.2         NaN   

  produced_by based_on distributed_by  
0         NaN      NaN            NaN

In [25]:
#Gestion nulos variables categoricas
def variables_categoricas(df):
   
    # Identificar columnas categóricas con valores nulos
    nulos_categoricas = df[df.columns[df.isnull().any()]].select_dtypes(include="O").columns
    print("Las columnas categóricas que tienen nulos son : \n")
    print(nulos_categoricas)

   # Asegurar que la columna 'release_date' esté en formato datetime
    df['release_date'] = pd.to_datetime(df['release_date'], errors='coerce')

    # Reemplazar los valores nulos en 'release_date' con la fecha estándar '1900-01-01'
    df['release_date'].fillna(pd.to_datetime('1900-01-01'), inplace=True)

    # Verificar la presencia de la fecha '1900-01-01'
    value_counts = df['release_date'].value_counts()
    display("Conteo de fechas en 'release_date':")
    display(value_counts)

    # Verificar que no queden valores nulos
    nulos_restantes = df['release_date'].isnull().sum()
    display(f"Valores nulos restantes en release_date: {nulos_restantes}")

    # Visualizar las primeras filas de la columna para ver los cambios
    display("Primeras filas de 'release_date':")
    display(df['release_date'].head(10))

    # Mostrar una muestra aleatoria de la columna para verificar
    display("Muestra aleatoria de 'release_date':")
    display(df['release_date'].sample(10))

#Llamada a la funcion
variables_categoricas(df)


Las columnas categóricas que tienen nulos son : 

Index(['release_date', 'directed_by', 'produced_by', 'based_on',
       'distributed_by'],
      dtype='object')


"Conteo de fechas en 'release_date':"

release_date
2016-08-12    36
1900-01-01    24
2011-04-22    20
2007-10-10    16
2004-06-13    16
              ..
2018-11-05     1
2019-06-11     1
1940-02-07     1
1937-12-21     1
1937-05-19     1
Name: count, Length: 422, dtype: int64

'Valores nulos restantes en release_date: 0'

"Primeras filas de 'release_date':"

0   1937-05-19
1   1937-12-21
2   1940-02-07
3   1940-11-13
4   1941-06-20
5   1941-10-23
6   1942-08-09
7   1942-08-24
7   1942-08-24
7   1942-08-24
Name: release_date, dtype: datetime64[ns]

"Muestra aleatoria de 'release_date':"

174   1990-10-03
363   2013-10-20
319   2007-10-10
224   1998-03-27
299   2007-03-23
59    1962-09-26
195   1994-10-28
116   1974-12-20
142   1979-12-21
334   2008-10-26
Name: release_date, dtype: datetime64[ns]

In [26]:
def variables_numericas(dg):
   
    # Identificar columnas numéricas con valores nulos
    nulos_numericas = df[df.columns[df.isnull().any()]].select_dtypes(include=np.number).columns
    print("Las columnas numéricas que tienen nulos son : \n")
    print(nulos_numericas)

    # Mostrar el porcentaje de valores nulos 
    print(df[nulos_numericas].isnull().sum() / df.shape[0] * 100)
   
    # Reemplazar nulos con la mediana
    columnas_mediana =['running_time','imdb']
    mediana = df[columnas_mediana].median()
    df[columnas_mediana] = df[columnas_mediana].fillna(mediana)
    
    # Verificar que los nulos en las columnas de mediana fueron reemplazados
    print("\nDespués de reemplazar nulos con la mediana:")
    print(df[columnas_mediana].isnull().sum())

    # Reemplazar nulos con fillna 0
    columnas_fillna_0=['budget', 'box_office']
    df[columnas_fillna_0] = df[columnas_fillna_0].fillna(0)

   # Verificar que los nulos en las columnas de fillna(0) fueron reemplazados
    print("\nDespués de reemplazar nulos con 0:")
    print(df[columnas_fillna_0].isnull().sum())

    # Copiar el DataFrame 
    df_copia = df.copy()
    
    # Guardar el DataFrame en un archivo CSV
    df_copia.to_csv("DisneyMoviesDataset_nonulls.csv", index=False)

    return df_copia

# Imputar valores nulos en variables numéricas
df_final = variables_numericas(df)
df_final.isnull().sum()

Las columnas numéricas que tienen nulos son : 

Index(['running_time', 'budget', 'box_office', 'imdb'], dtype='object')
running_time     2.772074
budget          25.872690
box_office      14.373717
imdb             4.004107
dtype: float64

Después de reemplazar nulos con la mediana:
running_time    0
imdb            0
dtype: int64

Después de reemplazar nulos con 0:
budget        0
box_office    0
dtype: int64


title                   0
production_company      0
country                 0
language                0
running_time            0
budget                  0
box_office              0
release_date            0
imdb                    0
directed_by             1
produced_by            21
based_on              560
distributed_by          9
dtype: int64

In [27]:
df_final.head()

title       production_company        country  \
0          Academy Award Review of  Walt Disney Productions  United States   
1  Snow White and the Seven Dwarfs  Walt Disney Productions  United States   
2                        Pinocchio  Walt Disney Productions  United States   
3                         Fantasia  Walt Disney Productions  United States   
4             The Reluctant Dragon  Walt Disney Productions  United States   

  language  running_time     budget   box_office release_date  imdb  \
0  English          41.0        0.0          0.0   1937-05-19   7.2   
1  English          83.0  1490000.0  418000000.0   1937-12-21   7.6   
2  English          88.0  2600000.0  164000000.0   1940-02-07   7.4   
3  English         126.0  2280000.0   83300000.0   1940-11-13   7.8   
4  English          74.0   600000.0     960000.0   1941-06-20   6.9   

                                         directed_by  \
0                                                NaN   
1  ['David Hand (supervising)', 'William Cottrell...   
2  ['Ben Sharpsteen', 'Hamilton Luske', 'Bill Rob...   
3  ['Samuel Armstrong', 'James Algar', 'Bill Robe...   
4  ['Alfred Werker', '(live action)', 'Hamilton L...   

                         produced_by  \
0                                NaN   
1                        Walt Disney   
2                        Walt Disney   
3  ['Walt Disney', 'Ben Sharpsteen']   
4                        Walt Disney   

                                            based_on  \
0                                                NaN   
1         ['Snow White', 'by The', 'Brothers Grimm']   
2  ['The Adventures of Pinocchio', 'by', 'Carlo C...   
3                                                NaN   
4                                                NaN   

                                      distributed_by  
0                                                NaN  
1                                 RKO Radio Pictures  
2                                 RKO Radio Pictures  
3  ['Walt Disney Productions', 'RKO Radio Pictures']  
4                                 RKO Radio Pictures